### ISSUE TITLE
[Feature] Adding a Color Code Calculator

### ISSUE THREAD

**Issue by SoumyadevSaha** [Rakesh9100/CalcDiverse#234]
### Description

I would like to add a new calculator to the list of calculators, which shall allow inter-conversion between hex color codes and RGB values. Eg. #2CAAB1 (hex code) is equal to rgb(44, 170, 177) in rgb format. Please assign me this issue under IWOC 2.0

### Checklist

- [X] I have checked the existing issues.
- [X] I follow [Contributing Guidelines](https://github.com/Rakesh9100/CalcDiverse/blob/main/.github/CONTRIBUTING_GUIDELINES.md) of this project.
- [X] I am a IWOC'24 contributor.
- [ ] I am a JWOC'24 contributor.
- [ ] I am a SWOC'24 contributor.

### Code of Conduct

- [X] I agree to follow this project's "[Code of Conduct](https://github.com/Rakesh9100/CalcDiverse/blob/main/.github/CODE_OF_CONDUCT.md)".

**Comment by Rakesh9100** [comment]
Pls edit the issue description and tick only the program you have created the issue for @SoumyadevSaha

**Comment by SoumyadevSaha** [comment]
> Pls edit the issue description and tick only the program you have created the issue for @SoumyadevSaha

Oh .. my bad .. extremely sorry for that. Edited the comment

**Comment by Rakesh9100** [comment]
Hey @SoumyadevSaha 

Instead of creating only RGB to Hex and vice versa, you can create a calculator named `Color Code Calculator` which includes the following:
1. RGB to RGBA and vice versa
2. RGB to Hex and vice versa
3. HSL to RGB and vice versa
4. HSV to RGB and vice versa
5. CMYK to RGB and vice versa
and so on!!

**Comment by SoumyadevSaha** [comment]
> Hey @SoumyadevSaha
> 
> Instead of creating only RGB to Hex and vice versa, you can create a calculator named `Color Code Calculator` which includes the following:
> 
> 1. RGB to RGBA and vice versa
> 2. RGB to Hex and vice versa
> 3. HSL to RGB and vice versa
> 4. HSV to RGB and vice versa
> 5. CMYK to RGB and vice versa
>    and so on!!

Sure, I can try that, but it might take some time, since along with coming up with a noce design format for the calculator, I also need to write accurate logics for these conversions, and also include checks for the correct format of inputs for each of these cases ... so, there is not any hard and fast deadline right ?

**Comment by SoumyadevSaha** [comment]
> > Hey @SoumyadevSaha
> > Instead of creating only RGB to Hex and vice versa, you can create a calculator named `Color Code Calculator` which includes the following:
> > 
> > 1. RGB to RGBA and vice versa
> > 2. RGB to Hex and vice versa
> > 3. HSL to RGB and vice versa
> > 4. HSV to RGB and vice versa
> > 5. CMYK to RGB and vice versa
> >    and so on!!
> 
> Sure, I can try that, but it might take some time, since along with coming up with a noce design format for the calculator, I also need to write accurate logics for these conversions, and also include checks for the correct format of inputs for each of these cases ... so, there is not any hard and fast deadline right ?

Also, since it might take a little time, shall I use this issue for JWOC which shall start on 15th or keep it as an issue under IWOC like it is now ?

**Comment by Rakesh9100** [comment]
> Sure, I can try that, but it might take some time, since along with coming up with a noce design format for the calculator, I also need to write accurate logics for these conversions, and also include checks for the correct format of inputs for each of these cases ... so, there is not any hard and fast deadline right ?

Yes, you have to write the logics for each conversion but don't worry, you have your time. For frontend, you can display 2 dropdowns in which codes are present for whichever conversion you will be doing and user can select from both the dropdowns and click on the button `Convert or Calculate` @SoumyadevSaha

**Comment by Rakesh9100** [comment]
> Also, since it might take a little time, shall I use this issue for JWOC which shall start on 15th or keep it as an issue under IWOC like it is now ? I am not sure about this, so it would be really nice if you could guide me on this matter.

It's totally your wish as per your availability and working if you want to keep it up for JWOC. I will keep it on hold and will assign when JWOC will start but only after you comment from 15th Jan @SoumyadevSaha

**Comment by Rakesh9100** [comment]
> Also, since it might take a little time, shall I use this issue for JWOC which shall start on 15th or keep it as an issue under IWOC like it is now ? I am not sure about this, so it would be really nice if you could guide me on this matter.

It's totally your wish as per your availability and working if you want to keep it up for JWOC. I will keep it on hold and will assign when JWOC will start but only after you comment from 15th Jan. So pls let me know further. @SoumyadevSaha

**Comment by Rakesh9100** [comment]
> Also, since it might take a little time, shall I use this issue for JWOC which shall start on 15th or keep it as an issue under IWOC like it is now ? I am not sure about this, so it would be really nice if you could guide me on this matter.

It's totally your wish as per your availability and working if you want to keep it up for JWOC. I will keep it on hold and will assign when JWOC will start but only after you comment from 15th Jan. So pls let me know further @SoumyadevSaha

**Comment by SoumyadevSaha** [comment]
> > Also, since it might take a little time, shall I use this issue for JWOC which shall start on 15th or keep it as an issue under IWOC like it is now ? I am not sure about this, so it would be really nice if you could guide me on this matter.
> 
> It's totally your wish as per your availability and working if you want to keep it up for JWOC. I will keep it on hold and will assign when JWOC will start but only after you comment from 15th Jan. So pls let me know further @SoumyadevSaha

I think I will keep it under IWOC as it it for now, since, I got some othe ideas for issues for JWOC, in the mean time I want to brainstorm those ideas. Please assign me this issue under IWOC @Rakesh9100

**Comment by SoumyadevSaha** [comment]
> > Sure, I can try that, but it might take some time, since along with coming up with a noce design format for the calculator, I also need to write accurate logics for these conversions, and also include checks for the correct format of inputs for each of these cases ... so, there is not any hard and fast deadline right ?
> 
> Yes, you have to write the logics for each conversion but don't worry, you have your time. For frontend, you can display 2 dropdowns in which codes are present for whichever conversion you will be doing and user can select from both the dropdowns and click on the button `Convert or Calculate` @SoumyadevSaha

Okay, then I will keep that as my front-end part design.

**Comment by Rakesh9100** [comment]
> I think I will keep it under IWOC as it it for now, since, I got some othe ideas for issues for JWOC, in the mean time I want to brainstorm those ideas. Please assign me this issue under IWOC @Rakesh9100

Sure, go ahead @SoumyadevSaha

**Comment by Sakshamjain98** [comment]
@Rakesh9100 sir please assign this issue to me under iwoc

**Comment by SoumyadevSaha** [comment]
Hi @Rakesh9100 made the PR for the color code calculator. Please do check.
